.github/workflows/check-generated-files.yaml

```diff
@@ -0,0 +1,25 @@
+name: check-generated-files
+on:
+  pull_request:
+    branches:    
+      - main
+
+jobs:
+  check-ui-static-files:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Generate static files for UI
+        run: make ui-static
+
+      - name: Verify Changed files
+        uses: tj-actions/verify-changed-files@v16
```

Comment:
I think it is recommended to pin untrusted (i.e. not published by actions, docker, or redhat-actions) third-party actions to their commit SHA instead: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions
I remember an internal email we received a while back about this.
But I noticed this is not done consistently in all our Workflows. We can therefore do need consistently in a subsequent PR if you prefer..